### PR TITLE
Add tx-status to cast

### PR DESCRIPTION
**Pull request by ksew1**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1986

## Introduced changes

- added `tx-status` subcommand for getting status of transaction

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/data/scripts/tx_status/src/status_succeeded.cairo [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use sncast_std::{tx_status, TxStatusResult, ExecutionStatus, FinalityStatus};
+
+fn main() {
+    let succeeded_tx_hash = 0x07d2067cd7675f88493a9d773b456c8d941457ecc2f6201d2fe6b0607daadfd1;
+    let status = tx_status(succeeded_tx_hash).unwrap();
+
+    println!("{}", status);
+    println!("{:?}", status);
+
```
Comment (Arcticae) [inline crates/sncast/tests/data/scripts/tx_status/src/status_succeeded.cairo:9]:
```suggestion
```

File: crates/sncast/tests/e2e/tx_status.rs [hunk 2]
```diff
@@ -17,7 +17,7 @@ async fn test_incorrect_transaction_hash() {
         output,
         indoc! {r"
         command: tx-status
-        error: Failed to get transaction status: TransactionHashNotFound
+        error: Failed to get transaction status: Transaction with provided hash was not found (does not exist)
```
Comment (Arcticae) [inline crates/sncast/tests/e2e/tx_status.rs:20]: Where does this change exactly come from?

File: docs/src/appendix/sncast-library.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 * [`invoke`](sncast-library/invoke.md) - invokes a contract's function
 * [`call`](sncast-library/call.md) - calls a contract's function
 * [`get_nonce`](sncast-library/get_nonce.md) - gets account's nonce for a given block tag
+* [`tx_status`](sncast-library/tx_status.md) - gets the status of a transaction using its hash
```
Comment (Arcticae) [inline docs/src/appendix/sncast-library.md:8]:
```suggestion
* [`tx_status`](sncast-library/tx_status.md) - gets the status of a transaction using its' hash
```

File: docs/src/appendix/sncast-library/tx_status.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+```rust
+use sncast_std::{tx_status};
+
+fn main() {
+    let transaction_hash = 0x00ae35dacba17cde62b8ceb12e3b18f4ab6e103fa2d5e3d9821cb9dc59d59a3c;
+    let status = tx_status(transaction_hash);
+
+    match status {
+        Result::Ok(TxStatusResult) => println!("transaction status: {:?}", TxStatusResult),
+        Result::Err(ScriptCommandError) => println!("error occurred: {:?}", ScriptCommandError),
```
Comment (Arcticae) [inline docs/src/appendix/sncast-library/tx_status.md:42]: It's better to assert here in case of error

File: sncast_std/src/lib.cairo [hunk 5]
```diff
@@ -282,6 +282,63 @@ pub fn get_nonce(block_tag: felt252) -> felt252 {
...
+}
+
+pub impl DisplayTxStatusResult of Display<TxStatusResult> {
+    fn fmt(self: @TxStatusResult, ref f: Formatter) -> Result<(), Error> {
+        let finality_status: ByteArray = match self.finality_status {
+            FinalityStatus::Received => "Received",
+            FinalityStatus::Rejected => "Rejected",
+            FinalityStatus::AcceptedOnL2 => "AcceptedOnL2",
+            FinalityStatus::AcceptedOnL1 => "AcceptedOnL1",
+        };
```
Comment (Arcticae) [inline sncast_std/src/lib.cairo:314]: This belongs in `FinalityStatus`'s `fmt` impl

File: sncast_std/src/lib.cairo [hunk 6]
```diff
@@ -282,6 +282,63 @@ pub fn get_nonce(block_tag: felt252) -> felt252 {
...
+        let execution_status = *self.execution_status;
+        if execution_status.is_none() {
+            return write!(f, "finality_status: {}", finality_status);
+        }
+
+        let execution_status: ByteArray = match execution_status.unwrap() {
+            ExecutionStatus::Succeeded => "Succeeded",
+            ExecutionStatus::Reverted => "Reverted"
+        };
```
Comment (Arcticae) [inline sncast_std/src/lib.cairo:323]: same here

**Review by Arcticae — COMMENTED** [review]
i have a few nitpicks, but apart from that it looks good 👌

File: docs/src/appendix/sncast-library/tx_status.md (hunk 4 above)
Comment (ksew1) [reply docs/src/appendix/sncast-library/tx_status.md]: I don't know if I understand correctly. You suggest to use expect?

File: crates/sncast/tests/e2e/tx_status.rs (hunk 2 above)
Comment (ksew1) [reply crates/sncast/tests/e2e/tx_status.rs:20]: I added [here](https://github.com/foundry-rs/starknet-foundry/blob/2644d6f00dd449f1f68d1d8c4f5fb8c63639f095/crates/sncast/src/starknet_commands/tx_status.rs#L23) conversion to StarknetCommandError. Which [has](https://github.com/foundry-rs/starknet-foundry/blob/7667a6ad819ebccd963810914a6ff9f54d5479be/crates/sncast/src/response/errors.rs#L66) custom display

**Review by Arcticae — APPROVED** [review]

**Pull request merged**